src/main/java/io/carbynestack/cli/client/castor/util/TupleFileParser.java

```diff
@@ -35,12 +38,32 @@ public static TupleChunk parse(TupleType tupleType, File tupleFile, UUID chunkId
 
     byte[] bytes;
     try (FileInputStream fileInputStream = new FileInputStream(tupleFile)) {
-      bytes = toByteArray(fileInputStream);
+      bytes = removeTupleHeader(fileInputStream);
     } catch (IOException e) {
       log.error(String.format("Cannot read file %s", tupleFile.getPath()));
       throw new RuntimeException();
     }
 
     return TupleChunk.of(tupleType.getTupleCls(), tupleType.getField(), chunkId, bytes);
   }
+
+  /**
+   * Removes the File Header that was introduced in MP-SPDZ v0.2.8
+   *
+   * @param tupleStreamWithHeader The file inputstream that includes the header
+   * @return The file bytes shifted by the header
+   */
+  private static byte[] removeTupleHeader(FileInputStream tupleStreamWithHeader)
```

Comment:
Q: Should this method be renamed as it does more than just remove the header?

Reply:
fixed